scripts/create-kubestellar-demo-env.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Deploys Kubestellar environment for demo purposes.
+
+set -e
+
+echo -e "Checking that pre-req softwares are installed..."
+curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v0.23.1/hack/check_pre_req.sh | bash -s -- -V kflex ocm helm kubectl docker kind
+
+output=$(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v0.23.1/hack/check_pre_req.sh | bash -s -- -A -V kflex ocm helm kubectl docker kind
+)
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+cluster_clean_up() {
+    error_message=$(eval "$1" 2>&1)
+    if [ $? -ne 0 ]; then
+        echo "clean up failed. Error:"
+        echo "$error_message"
+    fi
+}
+
+cluster_clean_up "kind delete cluster --name kubeflex"
+cluster_clean_up "kind delete cluster --name cluster1"
+cluster_clean_up "kind delete cluster --name cluster2"
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+
+context_clean_up() {
+    output=$(kubectl config get-contexts -o name)
+
+    while IFS= read -r line; do
```

Comment:
FYI: tather than indirect through a shell variable, this could be simpler: `kubectl config get-contexts -o name | while ... done`.

Reply:
I'll try that out